ztp/gitops-subscriptions/argocd/PowerConfiguration.md

```diff
@@ -0,0 +1,224 @@
+        profile:
+          - name: performance-patch
+            # The 'include' line must match the PerformanceProfile metadata.name above (openshift-node-performance-${metadata.name})
+            # And the cmdline_crash CPU set must match the 'isolated' set in the PerformanceProfile above
+            data: |
+              [main]
+              summary=Configuration changes profile inherited from performance created tuned
+              include=openshift-node-performance-openshift-node-performance-profile
+              [bootloader]
+              cmdline_crash=nohz_full=2-19,22-39
+              [sysctl]
+              kernel.timer_migration=1
+              [service]
+              service.stalld=start,enable
+```
+
+### 2) [High performance mode](#high-performance-mode) (ultra-low latency)
+
+Deploy the PerformanceProfile and TunedPerformancePatch policies as shown above with the addition of a new configuration under the `workloadHints` specification where `highPowerConsumption` is enabled.
+
+```yaml
+  sourceFiles:
+    - fileName: PerformanceProfile.yaml
+      policyName: "config-policy"
+      metadata:
+        name: openshift-node-performance-profile
+      spec:
+        ...
+        workloadHints:
+          highPowerConsumption: true
+        ...
+```
+
+### 3) [Power saving mode](#power-saving-mode)
+
+Deploy the PerformanceProfile and TunedPerformancePatch policies as shown below.
+
+Note that `highPowerConsumption` is disabled and `perPodPowerManagement` is enabled under the `workloadHints` specification. Both fields cannot be enabled simultaneously as this will result in an error. Furthermore, an additional kernel argument for the cpu governor is also added and set to `schedutil`.
+
+In order to maximize the power saving gain, the maximum cpu frequency should be capped. Without limiting the maximum cpu frequency, enabling C-states on the non-critical workload cpus allows the frequency of the critical cpus to be boosted, negating much of the power savings.
+
+The tuned performance-patch can be used to confine the maximum cpu frequency. The profile section of the performance-patch can be extended to use the `sysfs` plugin to set the `max_perf_pct`, which applies to all cpus. The `max_perf_pct` parameter controls the maximum frequency the cpufreq driver is allowed to set as a percent of the maximum supported cpu frequency (/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo\_max\_freq). This is demonstrated in the example below.
+
+> **Note** `/sys/devices/system/cpu/intel_pstate/max_perf_pct=x` where `x` is to be configured based on the hardware specification and powersaving requirements.
+
+```yaml
+  sourceFiles:
+    - fileName: PerformanceProfile.yaml
+      policyName: "config-policy"
+      metadata:
+        name: openshift-node-performance-profile
+      spec:
+        additionalKernelArgs:
+          - "cpufreq.default_governor=schedutil"
+          - "rcupdate.rcu_normal_after_boot=0"
+          - "efi=runtime"
```

Comment:
This is going to get out of date. Probably need to add a comment about where the list should be taken from.